ckanext/canada/tests/test_xls_upload.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: UTF-8 -*-
+import os
+from nose.tools import assert_equal, assert_raises
+from ckanapi import LocalCKAN
+
+from ckan.tests.helpers import FunctionalTestBase
+from ckan.tests.factories import Organization
+
+from ckanext.recombinant.tables import get_chromo, get_geno
+from ckanext.recombinant.controller import _process_upload_file
+from ckanext.recombinant.write_excel import excel_template, append_data
+from ckanext.recombinant.errors import  BadExcelData
+from openpyxl import load_workbook, Workbook
+
+
+CURRENT_PATH = os.path.dirname(os.path.abspath(__file__))
```

Comment:
use the `tempfile` module to securely create temporary files/directories instead of using the source directory

Reply:
Thanks! I just implemented that in the new commit.